arlas-core/src/main/java/io/arlas/server/core/impl/elastic/services/ElasticFluidSearch.java

```diff
@@ -622,302 +615,291 @@ private Builder.ContainerBuilder buildDateHistogramAggregation(Aggregation aggre
+            if (hasSameSignedSort(rg.sort, signedFetchIncludes)) {
+                rg.setSignedSort(String.join(",", signedFetchIncludes));
+                rg.setInclude(aggregationModel.fetchHits.include);
+                mergeableRS.add(rg);
+            }
         }
-        if(aggregationModel.rawGeometries == null && aggregationModel.fetchHits != null){
-            return this.setHitsToFetch(aggregationModel,containerBuilder);
+
+        if (mergeableRS.isEmpty() || fetchSize != 1) {
+            setRawGeometries(aggregationModel,containerBuilder);
+            setHitsToFetch(aggregationModel,containerBuilder);
+            return containerBuilder;
         }
-        //If no rawGeometries and no fetchHits
-        return  containerBuilder;
+
+        return mergeIntoTopHits(containerBuilder, mergeableRS);
     }
 
+    private List<String> getSignedFetchIncludes(List<String> includes) throws ArlasException {
+        if (includes == null) return Collections.emptyList();
+        List<String> signedIncludes = new ArrayList<>();
+        for (String field : includes) {
+            String unsignedField = field.replaceFirst(ORDER_SIGN_REGEX, "");
+            CollectionUtil.checkAliasMappingFields(client.getMappings(collectionReference.params.indexName), unsignedField);
+            if (field.startsWith("+") || field.startsWith("-")) {
+                signedIncludes.add(field);
+            }
+        }
+        return signedIncludes;
+    }
+    private boolean hasSameSignedSort(String sort, List<String> signedIncludes) {
+        List<String> signedSortFields = Arrays.stream(sort.split(","))
+                .filter(f -> f.startsWith("+") || f.startsWith("-"))
+                .toList();
+        return String.join(",", signedSortFields).equals(String.join(",", signedIncludes));
+    }
 
+    private Builder.ContainerBuilder mergeIntoTopHits(Builder.ContainerBuilder containerBuilder, List<RawGeometry> mergeableRS) throws ArlasException {
+        Map<String, Set<String>> rgs = new HashMap<>();
+        for (RawGeometry rg : mergeableRS) {
+            rgs.computeIfAbsent(rg.signedSort, k -> new HashSet<>()).addAll(rg.include);
+            rgs.get(rg.signedSort).add(rg.geometry);
+        }
+        for (Map.Entry<String, Set<String>> entry : rgs.entrySet()) {
+            String sort = entry.getKey();
+            String[] includes = entry.getValue().toArray(String[]::new);
+            TopHitsAggregation.Builder topHitsBuilder = AggregationBuilders.topHits().size(1)
+                    .source(s -> s.filter(f -> f.includes(Arrays.asList(includes))));
+            for (String field : sort.split(",")) {
+                String unsignedField = field.replaceFirst(ORDER_SIGN_REGEX, "");
+                CollectionUtil.checkAliasMappingFields(client.getMappings(collectionReference.params.indexName), unsignedField);
+                SortOrder order = field.startsWith("+") ? SortOrder.Asc : SortOrder.Desc;
```

Comment:
I checked the doc for fertch hits and we apply the order only if the field is preceded by + or -, in case we don't add a sign, no order is to be applied